### PR TITLE
Add JVM runtime metric source API notes

### LINKED ISSUE #3418: Add link to related Java API(s) in note section for all JVM metrics
I think this is helpful additional information for the JVM metrics and would help implementers.

### PR DESCRIPTION
**Pull request by roberttoyonaga**
Fixes #3418

## Changes

Add notes on where the JVM runtime metrics come from.

### CODE REVIEW

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 1]
```diff
@@ -297,6 +309,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.system.cpu.utilization`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [OperatingSystemMXBean#getSystemCpuLoad()](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getSystemCpuLoad()).
```
Comment (breedx-splk) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:312]: This one [depends on the version of java](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/Cpu.java#L58) and is unfortunately more complicated. I think it's worth providing the details:

* `com.sun.management.OperatingSystemMXBean.getCpuLoad()` with 14+
* `com.sun.management.OperatingSystemMXBean.getSystemCpuLoad()` for 8-13
* `com.ibm.lang.management.OperatingSystemMXBean.getSystemCpuLoad()` for j9

**Review by breedx-splk — APPROVED** [review]
Looks good. 👍🏻 Offered a small idea on `process.runtime.jvm.system.cpu.utilization`.

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md (hunk 1 above)
Comment (roberttoyonaga) [reply specification/metrics/semantic_conventions/runtime-environment-metrics.md]: thanks! That's a good point. I've added that info.

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ consider, for example pthreads vs green thread implementations.
 ### Metric: `process.runtime.jvm.memory.usage`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [MemoryPoolMXBean#getUsage()](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--).
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:81]: Nit: code formatting
```suggestion
This metric is obtained from [`MemoryPoolMXBean#getUsage()`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--).
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 3]
```diff
@@ -104,6 +105,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.memory.init`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [MemoryPoolMXBean#getUsage()](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--).
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:108]:
```suggestion
This metric is obtained from [`MemoryPoolMXBean#getUsage()`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--).
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 4]
```diff
@@ -130,6 +132,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.memory.committed`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [MemoryPoolMXBean#getUsage()](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--).
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:135]:
```suggestion
This metric is obtained from [`MemoryPoolMXBean#getUsage()`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--).
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 5]
```diff
@@ -156,6 +159,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.memory.limit`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [MemoryPoolMXBean#getUsage()](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--).
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:162]:
```suggestion
This metric is obtained from [`MemoryPoolMXBean#getUsage()`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getUsage--).
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 6]
```diff
@@ -182,6 +186,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.memory.usage_after_last_gc`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [MemoryPoolMXBean#getCollectionUsage()](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getCollectionUsage--).
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:189]:
```suggestion
This metric is obtained from [`MemoryPoolMXBean#getCollectionUsage()`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryPoolMXBean.html#getCollectionUsage--).
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 7]
```diff
@@ -271,6 +281,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.classes.current_loaded`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [ClassLoadingMXBean#getLoadedClassCount()](https://docs.oracle.com/javase/8/docs/api/java/lang/management/ClassLoadingMXBean.html#getLoadedClassCount--).
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:284]:
```suggestion
This metric is obtained from [`ClassLoadingMXBean#getLoadedClassCount()`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/ClassLoadingMXBean.html#getLoadedClassCount--).
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 8]
```diff
@@ -284,6 +295,8 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.cpu.utilization`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [com.sun.management.OperatingSystemMXBean#getProcessCpuLoad()](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getProcessCpuLoad()) on HotSpot
+and [com.ibm.lang.management.OperatingSystemMXBean#getCpuLoad()](https://www.ibm.com/docs/hr/sdk-java-technology/8?topic=SSYKE2_8.0.0/openj9/api/jdk8/jre/management/extension/com/ibm/lang/management/OperatingSystemMXBean.htm) on J9.
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:299]:
```suggestion
This metric is obtained from [`com.sun.management.OperatingSystemMXBean#getProcessCpuLoad()`](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getProcessCpuLoad()) on HotSpot
and [`com.ibm.lang.management.OperatingSystemMXBean#getCpuLoad()`](https://www.ibm.com/docs/hr/sdk-java-technology/8?topic=SSYKE2_8.0.0/openj9/api/jdk8/jre/management/extension/com/ibm/lang/management/OperatingSystemMXBean.htm) on J9.
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 9]
```diff
@@ -297,6 +310,9 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.system.cpu.utilization`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [com.sun.management.OperatingSystemMXBean#getSystemCpuLoad()](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getSystemCpuLoad()) on Java version 8..13, [com.sun.management.OperatingSystemMXBean#getCpuLoad()](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getCpuLoad()) on Java version 14+,
+and [com.ibm.lang.management.OperatingSystemMXBean#getCpuLoad()](https://www.ibm.com/docs/hr/sdk-java-technology/8?topic=SSYKE2_8.0.0/openj9/api/jdk8/jre/management/extension/com/ibm/lang/management/OperatingSystemMXBean.htm) on J9.
+
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:315]:
```suggestion
This metric is obtained from [`com.sun.management.OperatingSystemMXBean#getSystemCpuLoad()`](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getSystemCpuLoad()) on Java version 8..13, [`com.sun.management.OperatingSystemMXBean#getCpuLoad()`](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getCpuLoad()) on Java version 14+,
and [`com.ibm.lang.management.OperatingSystemMXBean#getCpuLoad()`](https://www.ibm.com/docs/hr/sdk-java-technology/8?topic=SSYKE2_8.0.0/openj9/api/jdk8/jre/management/extension/com/ibm/lang/management/OperatingSystemMXBean.htm) on J9.
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 10]
```diff
@@ -323,6 +340,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.buffer.usage`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [BufferPoolMXBean#getMemoryUsed()](https://docs.oracle.com/javase/8/docs/api/java/lang/management/BufferPoolMXBean.html#getMemoryUsed--).
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:343]:
```suggestion
This metric is obtained from [`BufferPoolMXBean#getMemoryUsed()`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/BufferPoolMXBean.html#getMemoryUsed--).
```

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 11]
```diff
@@ -341,6 +359,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.buffer.limit`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [BufferPoolMXBean#getTotalCapacity()](https://docs.oracle.com/javase/8/docs/api/java/lang/management/BufferPoolMXBean.html#getTotalCapacity--).
```
Comment (mateuszrzeszutek) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:362]:
```suggestion
This metric is obtained from [`BufferPoolMXBean#getTotalCapacity()`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/BufferPoolMXBean.html#getTotalCapacity--).
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 12]
```diff
@@ -301,6 +314,8 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.system.cpu.utilization`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [`com.sun.management.OperatingSystemMXBean#getSystemCpuLoad()`](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getSystemCpuLoad()) on Java version 8..13, [`com.sun.management.OperatingSystemMXBean#getCpuLoad()`](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.management/com/sun/management/OperatingSystemMXBean.html#getCpuLoad()) on Java version 14+,
+and [`com.ibm.lang.management.OperatingSystemMXBean#getCpuLoad()`](https://www.ibm.com/docs/hr/sdk-java-technology/8?topic=SSYKE2_8.0.0/openj9/api/jdk8/jre/management/extension/com/ibm/lang/management/OperatingSystemMXBean.htm) on J9.
```
Comment (trask) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:318]: let's find language-neutral link

Comment (roberttoyonaga) [reply specification/metrics/semantic_conventions/runtime-environment-metrics.md]: Ok I've updated those links

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 13]
```diff
@@ -208,6 +213,7 @@ This metric is [recommended](../metric-requirement-level.md#recommended).
 ### Metric: `process.runtime.jvm.gc.duration`
 
 This metric is [recommended](../metric-requirement-level.md#recommended).
+This metric is obtained from [`GarbageCollectorMXBean`](https://docs.oracle.com/javase/8/docs/api/java/lang/management/GarbageCollectorMXBean.html).
```
[the remark this answers is not in the document]
Comment (roberttoyonaga) [reply specification/metrics/semantic_conventions/runtime-environment-metrics.md]: Good idea. I've added that extra context.

**Comment by jack-berg** [comment]
@roberttoyonaga can you add a changelog entry? Sorry I forgot mention this before.

**Pull request merged**